eclipse/src/saros/ui/wizards/pages/EnterProjectNamePage.java

```diff
@@ -371,11 +382,27 @@ private void preselectProjectNames() {
 
       if (project != null) continue;
 
-      String projectNameProposal =
-          findProjectNameProposal(
-              remoteProjectMapping.get(projectID), reservedProjectNames.toArray(new String[0]));
+      final String remoteProjectName = remoteProjectIdToNameMapping.get(projectID);
+
+      final String desiredLocalProjectName =
+          desiredRemoteToLocalProjectNameMapping.get(remoteProjectName);
+
+      boolean existingProject = false;
+
+      String projectNameProposal = null;
+
+      if (desiredLocalProjectName != null
+          && !reservedProjectNames.contains(desiredLocalProjectName)) {
```

Comment:
I am a bit confused by the usage of `projectOptionComposites`. `reservedProjectNames` are filled using `projectOptionComposites`, but you also use it to iterate over the projects to suggest proposals for.

What does the map `projectOptionComposites` contain? The mapping of already shared projects? I.e. are the names 'reserved' as there already is a mapping for the project in the current session?

Reply:
https://help.eclipse.org/kepler/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fswt%2Fwidgets%2FComposite.html


as Swing https://docs.oracle.com/javase/7/docs/api/javax/swing/JComponent.html

Actually this is the content of a tab.

I know this class is a mess but it were much more worse years ago.

Reply:
Ok, I did not see that the two loops run on different continue conditions. Thought that both were checking `project==null`.

How about adding a short comment before each loop stating the purpose as this is really hard to understand otherwise? This would be a compromise to moving them to different method (even though I also prefer to split this into different methods, but I get your argument against it; as a side note: data re-creation could be avoided by returning the reserved names from the first method and passing it to the second).